Sources/XMLCoder/Auxiliaries/XMLStackParser.swift

```diff
@@ -141,15 +145,38 @@ extension XMLStackParser: XMLParserDelegate {
             return
         }
 
+        let updatedElement = removeWhitespaceElements ? elementWithFilteredElements(element: element) : element
+    
         withCurrentElement { currentElement in
-            currentElement.append(element: element, forKey: element.key)
+            currentElement.append(element: updatedElement, forKey: updatedElement.key)
         }
 
         if stack.isEmpty {
-            root = element
+            root = updatedElement
         }
     }
 
+    func elementWithFilteredElements(element: XMLCoderElement) -> XMLCoderElement {
+        var hasWhitespaceElements: Bool = false
+        var hasNonWhitespaceElements: Bool = false
+        var filteredElements: [XMLCoderElement] = []
+        for ele in element.elements {
+            if ele.isWhitespaceWithNoElements() {
+                hasWhitespaceElements = true
+            } else {
+                hasNonWhitespaceElements = true
+                filteredElements.append(ele)
+            }
+        }
+        let updatedElement: XMLCoderElement
+        if hasWhitespaceElements && hasNonWhitespaceElements {
+            updatedElement = XMLCoderElement(key: element.key, elements: filteredElements, attributes: element.attributes)
```

Comment:
Question. Is it possible to just return from this if else statement instead of setting a variable?
```swift
if hasWhitespaceElements && hasNonWhitespaceElements {
            return XMLCoderElement(key: element.key, elements: filteredElements, attributes: element.attributes)
 } 

return element
```

Reply:
Ah, sure, thanks!